src/unix/stream.c

```diff
@@ -19,6 +19,19 @@
  * IN THE SOFTWARE.
  */
 
+/* Not all types of file descriptors work with poll(2) and kqueue(2),
+ * presumably because the richest company on Earth couldn't allocate
+ * resources to add support for them. Programming is hard, we understand.
+ * Libuv detects and relegates such file descriptors to a separate handler
+ * thread that polls them like it's 1979 again using select(2) but fd_sets
+ * have an inescapable upper bound. We increase said bound and hope and
+ * assume that's enough for everyday use but it remains a sad fact of life
+ * that some file descriptors won't work on macOS.
+ */
+#if defined(__APPLE__)
+#define FD_SETSIZE 32768
+#endif
```

Comment:
#define _DARWIN_UNLIMITED_SELECT

(and remove the error code later, as this is available since 10.5 at least)

Reply:
That stirred a memory.... we already build with `-D_DARWIN_UNLIMITED_SELECT=1`! Now I'm left wondering if the ubsan warning from #4262 is a false positive or we're somehow losing the build flag.